portal/src/test/scala/cromwell/pipeline/utils/golden/ResourceFileCodecLaws.scala

```diff
@@ -0,0 +1,108 @@
+package cromwell.pipeline.utils.golden
+
+import cats.instances.list._
+import cats.instances.try_._
+import cats.syntax.apply._
+import cats.syntax.traverse._
+import java.io.File
+import java.io.PrintWriter
+
+import org.scalacheck.{Arbitrary, Gen}
+import play.api.libs.json.{Format, JsResult, JsValue}
+
+import scala.io.{BufferedSource, Source}
+import scala.reflect.runtime.universe.TypeTag
+import scala.util.{Failure, Success, Try}
+
+abstract class ResourceFileCodecLaws[A](
+  name: String,
+  resourceRootDir: File,
+  resourcePackage: List[String],
+  val size: Int = 100,
+  count: Int = 1
+) extends CodecLaws[A]
+    with SampleGeneration[A] {
+
+  protected lazy val goldenObjects: Try[List[(A, String)]] =
+    loadGoldenFiles match {
+      case Success(value) if value.isEmpty => generateGoldenFiles
+      case Failure(_)                      => generateGoldenFiles
+      case _                               => loadGoldenFiles
+    }
+
+  private val rootPath = "/" + resourcePackage.mkString("/") + "/"
+  private val resourceDir: File = resourcePackage.foldLeft(resourceRootDir) {
+    case (acc, p) => new File(acc, p)
+  }
+  private val goldenFileNamePattern = s"d^$name-(.{50})\\.json$$".r
+
+  private lazy val loadGoldenFiles = {
+    Option(getClass.getResourceAsStream(rootPath)) match {
+      case Some(inputStream) =>
+        val resource = Source.fromInputStream(inputStream)
+        open(resource) { dirSource =>
+          val files = dirSource.getLines.flatMap {
+            case fileName @ goldenFileNamePattern(seed) => Some((seed, fileName))
+            case _                                      => None
+          }.toList.traverse[Try, (A, String)] {
+            case (seed, name) =>
+              val contents = Resources.open(rootPath + name).map { source =>

```

Comment:
I think you should use here the new `open` method that can close resources automatically.

Reply:
That `open` method is used there. I removed the other from the `Resources` object.

Reply:
But now you have two `open` methods. One of them closes resources safely and the other one does not. This may cause some problems. I propose to deduplicate these methods and leave only one method that will safely close the resource.

Reply:
You are right. I thought I removed the other `open` method (which doesn't close the source). I'll do it right away.